### DISCUSSION TITLE
What does the "peak" memory consumption mean?

### DISCUSSION THREAD

**Discussion by rongardF** [plasma-umass/scalene#618] [Q&A] [answered]
I am running a docker container with memory limit set to 6GB. Inside this container there is a Python program executed using Scalene profiling tool. In the results there is a line which is indicated to have "peak" memory consumption of 13.4GB. This should not be possible because the container should get killed when exceeding 6GB. So what does Scalene "peak" memory consumption mean and how it is derived? Is it perhaps the accumulative memory consumption of the entire program execution?

![image](https://github.com/plasma-umass/scalene/assets/42897796/6ac09921-f813-4eb8-8b70-d1d3f3dc129a)

**Comment by emeryberger** [comment, marked as answer]
The peak memory is the peak _allocated_ (requested by the app) but not necessarily the maximum amount _used_. If you allocate a very large object, it is allocated via `mmap`, which does not consume (physical) memory until the memory inside the object is actually accessed.

**Reply by rongardF** [reply]
Okay, thank you for the clarification!

**Comment by ogencoglu** [comment]
Thanks for the clarification. Is there a way to know the peak _used_ memory then? @emeryberger
